internal/session/sessionwantsender.go

```diff
@@ -215,13 +234,20 @@ func (sws *sessionWantSender) onChange(changes []change) {
 
 	// Apply each change
 	availability := make(map[peer.ID]bool, len(changes))
+	cancels := cid.NewSet()
 	var updates []update
 	for _, chng := range changes {
 		// Initialize info for new wants
 		for _, c := range chng.add {
 			sws.trackWant(c)
 		}
 
+		// Remove cancelled wants
+		for _, c := range chng.cancel {
+			sws.untrackWant(c)
```

Comment:
Can we check to see if it was wanted in the first place? That should also let us use a simple slice instead of a set (no duplicates).

Reply:
The reason to pass cancels through the `sessionWantSender` is just to make sure that cancels come after wants. But I think we probably want the `PeerManager` be the authority on whether a want has actually been sent.

I agree that we can use a slice instead of a set, I will update.